concepts/functions/about.md

```diff
@@ -1,3 +1,176 @@
 # About
 
-TODO
+Functions are bodies of code containing one or more [statements or expressions][statements or expressions]. A function may optionally
+[return an expression][return value]. (Recall that a statement is an instruction that does not return a value. An expression evaluates to
+a resulting value.)
+
+Rust style is to use [snake case][snake case] for a function name, which is prefaced by the `fn` keyword for a function definition. For example
+
+```rust
+fn do_nothing() {}
+
+```
+
+The empty parentheses are a placeholder for zero or more parameters. A parameter is the definition of a value to be passed into the function.
+Even if there are no parameters the parentheses are still required in the function definition. The actual value passed to a parameter is called
+an argument.
+
+The combination of parameters and return value is known as the function's signature. A function's signature requires that each parameter must
+have its type annotated.
+
+The curly braces enclose the body of the function. Inside the braces are where the function's statements and expressions are placed. Even if the
+function contains no statements or expressions the curly braces are still required for the function definition. Of course there is no point to
+having an empty function.
+
+A function could contain a simple statement. In the following example the function takes in one `i32` argument and prints it
+
+```rust
+fn print_integer(value: i32) {
+    println!("{:?}", value);
+}
+```
+
+Note that the parameter needs to be given a name (`value` in this case) and be annotated with its type.
+
+A function can also return an expression. In the following example the function takes in one `i32` argument and returns it doubled
+
+```rust
+fn double_integer(value: i32) -> i32 {
+    value * 2
+}
+```
+
+The `-> i32` indicates that the function returns an `i32`. Unlike parameters, the returned value is not named. Note that the last expression
+in a function can be returned without the `return` keyword and a semicolon. `return` _is_ needed to return early from a function, like so
```

Comment:
Two concepts, two paragraphs.

```suggestion
The `-> i32` indicates that the function returns an `i32`. Unlike parameters, the returned value is not named. 

It is possible to exit from a function early with the `return` keyword, like so:
```

I want to instill in students the sense that the normal, expected pattern is to terminate a function with an expression returning the final value, and that the `return` keyword should only be used to exit early. That affects how we write about it.